examples/beacon-api-sidecar-fetcher/src/main.rs

```diff
@@ -0,0 +1,235 @@
+//! Run with
+//!
+//! ```not_rust
+//! cargo run -p beacon-api-sidecar-fetcher -- node
+
+use std::{
+    collections::VecDeque,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use thiserror::Error;
+
+use futures_util::{stream::FuturesUnordered, Future, Stream, StreamExt};
+use reqwest::{Error, StatusCode};
+use reth::{
+    primitives::{alloy_primitives::TxHash, BlobTransactionSidecar, BlockHash, BlockNumber},
+    providers::CanonStateNotification,
+    rpc::types::engine::BlobsBundleV1,
+    transaction_pool::{BlobStoreError, TransactionPoolExt},
+};
+
+#[tokio::main]
+async fn main() -> eyre::Result<()> {
+    Ok(())
+}
+
+//TODO Add nicer errors.
+#[derive(Debug, Error)]
+pub enum SideCarError {
+    #[error("Reqwest encountered an error: {0}")]
+    ReqwestError(Error),
+
+    #[error("There was an error grabbing the blob from the tx pool: {0}")]
+    TransactionPoolError(BlobStoreError),
+
+    #[error("400: {0}")]
+    InvalidBlockID(String),
+
+    #[error("404: {0}")]
+    BlockNotFound(String),
+
+    #[error("500: {0}")]
+    InternalError(String),
+
+    #[error("Network error: {0}")]
+    NetworkError(String),
+
+    #[error("Data parsing error: {0}")]
+    DeserializationError(String),
+
+    #[error("{0} Error: {1}")]
+    UnknownError(u16, String),
+}
+
+#[derive(Debug, Clone)]
+pub struct MinedSidecar {
+    tx_hash: TxHash,
+    block_hash: BlockHash,
+    block_number: BlockNumber,
+    sidecar: BlobTransactionSidecar,
```

Comment:
this should be

https://github.com/paradigmxyz/reth/blob/250da4e449d1872625fa6d8b0a943ceb0f6b8379/crates/primitives/src/transaction/sidecar.rs#L53-L53

basically the mined txs reattached with its sidecar